tests/ui/useless_vec.stderr

```diff
@@ -0,0 +1,16 @@
+error: useless use of `vec!`
+  --> tests/ui/useless_vec.rs:8:26
+   |
+LL |       let _some_variable = vec![
+   |  __________________________^
+LL | |
+LL | |         1, 2, // i'm here to stay
+LL | |         3, 4, // but this one going away ;-;
+LL | |     ]; // that is life anyways
```

Comment:
IMO the help message/suggestion was a fairly important part of the lint because now it's not 100% clear what the lint wants the user to do from the diagnostic alone (use an array instead).

What about just changing the applicability to something else other than `MachineApplicable` so that `--fix` won't autofix it but is still shown to the user as a hint to what the fix roughly looks like and they can apply it manually by preserving the comment?

(I guess ideally the suggestion could just include the last comment directly and keep it as `MachineApplicable` since all other comments except for the one after the last element are already correctly handled, but I can only think of hacky ways to do it like extending the snippet's span to the full `vec![]` invocation minus 1 (just behind the `]`), and am not sure if that will always work so 🤷)

Reply:
Yeah makes sense. Gonna change the applicability instead.